packages/eas-cli/src/commands/update/index.ts

```diff
@@ -577,6 +602,38 @@ export default class UpdatePublish extends EasCommand {
       }
     }
   }
+
+  private sanitizeFlags(flags: RawUpdateFlags): UpdateFlags {
+    const nonInteractive = flags['non-interactive'] ?? false;
+
+    const { auto, branch: branchName, message: updateMessage } = flags;
+    if (nonInteractive && !auto && !(branchName && updateMessage)) {
+      Errors.error(
```

Comment:
these exit/error methods (that return `never`) are pretty oddly-typed. For example, if you pass `{exit: false}` in as the second argument, it doesn't exit and the typescript types are incorrect.

I made a similar comment on another recent PR (https://github.com/expo/eas-cli/pull/1427#discussion_r994887505) but my general feeling is that these add a bit of overhead for the reviewer since it's not immediately clear just from reading that they halt the procedure (like a throw would). That being said, I see that these are used in this param sanitization pretty consistently, so I guess it's okay here. It still took me a bit longer to review though, so just a data point.